homeassistant/components/sensor/viaggiatreno.py

```diff
@@ -0,0 +1,137 @@
+"""
+Trenitalia sensor.
+
+This simple sensor uses ViaggiaTreno APIs to provide information about Italian public railroads
+status.
+"""
+import datetime
+import logging
+import urllib.request
+import json
+import re
+import voluptuous as vol
+from urllib.parse import quote, urljoin
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.util import Throttle
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = []
+
+log = logging.getLogger(__name__)
+
+CONF_STATION_NAME = 'station_name'
+CONF_TRAIN_NO = 'train_no'
+CONF_ORIGIN_STATION = 'origin_station'
+CONF_ATTRIBUTION = 'Data provided by ViaggiaTreno.it'
+ICON = 'mdi:train'
+RESOURCE_URL = "http://www.viaggiatreno.it/viaggiatrenonew/resteasy/viaggiatreno/"
+
+MIN_TIME_BETWEEN_UPDATES = datetime.timedelta(seconds=120)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STATION_NAME): cv.string,
+    vol.Required(CONF_TRAIN_NO): cv.string,
+    vol.Optional(CONF_ORIGIN_STATION): cv.string,
+})
+
+def get_origin_station(train_no):
+    '''
+    Get station id from train number.
+    '''
+    url = urljoin(RESOURCE_URL, "cercaNumeroTrenoTrenoAutocomplete/{}".format(train_no))
+    r = urllib.request.urlopen(url)
```

Comment:
Most platforms are using `requests` to interact with third party services. Please add a timeout to the request.